[receiver/httpcheck] add metrics check whether substring contains in response body (#27015)
open-telemetry/opentelemetry-collector-contrib: receiver/httpcheckreceiver/README.md

- `endpoint` (required): the URL to be monitored
- `method` (optional, default: `GET`): The HTTP method used to call the endpoint

> - `substring` (optional, default: ""): The substring contains in response body of the HTTP check

Reviewer: ```suggestion
- `substring` (optional, default: ""): If set, the receiver will emit metrics based on whether the response body matches this string.
```

Author: Done

> - `substring` (optional, default: ""): If set, the receiver will emit metrics based on whether the response body matches this string.

Reviewer 2: As per the description of this option, `body` seems more descriptive

```suggestion
- `body` (optional, default: ""): If set, the receiver will emit metrics based on whether the response body matches this string.
```

Reviewer 3: I agree `body` is much better name for both the configuration property and the metric name.

I would also expect the description to specify whether it's a regex match or exact match.

- `endpoint` (required): the URL to be monitored
- `method` (optional, default: `GET`): The HTTP method used to call the endpoint

> - `body` (optional, default: ""): If set, the receiver will emit metrics based on whether the response body exact matches this string.

Reviewer 4: This could carry a different meaning when method is `POST` where the `body` content will be sent as part of the request.

> - `name` (optional, default: ""): If set, along with `httpcheck.body` metric attributes.

Reviewer: ```suggestion
- `name` (optional, default: ""): The name of this receiver.
```

Author: Done
